packages/x-data-grid/src/models/api/gridVirtualizationApi.ts

```diff
@@ -17,4 +17,8 @@ export interface GridVirtualizationPrivateApi {
    * @returns {GridRenderContext} The `GridRenderContext`.
    */
   updateRenderContext?: () => void;
+  /**
+   * Lock/unlock the virtual scroller events.
+   */
+  setVirtualScrollerLock: (locked: boolean) => void;
```

Comment:
The virtualization logic was running during resize in one edge case, it was too complicated to solve it otherwise so I've just prevented it from running. This is terrible but I think it's the best solution.